File: 4_data_structures_and_libraries/1763.py
# https://www.urionlinejudge.com.br/judge/en/problems/view/1763

from sys import stdin
from collections import defaultdict

pairs = (
    ("brasil", "Feliz Natal!"),
    ("alemanha", "Frohliche Weihnachten!"),
    ("austria", "Frohe Weihnacht!"),
    ("coreia", "Chuk Sung Tan!"),
    ("espanha", "Feliz Navidad!"),
    ("grecia", "Kala Christougena!"),
    ("estados-unidos", "Merry Christmas!"),
    ("inglaterra", "Merry Christmas!"),
    ("australia", "Merry Christmas!"),
    ("portugal", "Feliz Natal!"),
    ("suecia", "God Jul!"),
    ("turquia", "Mutlu Noeller"),
    ("argentina", "Feliz Navidad!"),
    ("chile", "Feliz Navidad!"),
    ("mexico", "Feliz Navidad!"),
    ("antardida", "Merry Christmas!"),
    ("canada", "Merry Christmas!"),
    ("irlanda", "Nollaig Shona Dhuit!"),
    ("belgica", "Zalig Kerstfeest!"),
    ("italia", "Buon Natale!"),
    ("libia", "Buon Natale!"),
    ("siria", "Milad Mubarak!"),
    ("marrocos", "Milad Mubarak!"),
    ("japao", "Merii Kurisumasu!"))

dd = dict(pairs)

while True:
    try:
        country = input()
    except EOFError:
        break
    else:
        try:
            message = dd[country]
        except KeyError:
            print("--- NOT FOUND ---")
        else:
            print(message)

File: 4_data_structures_and_libraries/1763.cpp
// https://www.urionlinejudge.com.br/judge/en/problems/view/1763

#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{
    string input;
    map<string, string> dict;
    map<string, string>::iterator it;

    dict["brasil"] = "Feliz Natal!";
    dict["alemanha"] = "Frohliche Weihnachten!";
    dict["austria"] = "Frohe Weihnacht!";
    dict["coreia"] = "Chuk Sung Tan!";
    dict["espanha"] = "Feliz Navidad!";
    dict["grecia"] = "Kala Christougena!";
    dict["estados-unidos"] = "Merry Christmas!";
    dict["inglaterra"] = "Merry Christmas!";
    dict["australia"] = "Merry Christmas!";
    dict["portugal"] = "Feliz Natal!";
    dict["suecia"] = "God Jul!";
    dict["turquia"] = "Mutlu Noeller";
    dict["argentina"] = "Feliz Navidad!";
    dict["chile"] = "Feliz Navidad!";
    dict["mexico"] = "Feliz Navidad!";
    dict["antardida"] = "Merry Christmas!";
    dict["canada"] = "Merry Christmas!";
    dict["irlanda"] = "Nollaig Shona Dhuit!";
    dict["belgica"] = "Zalig Kerstfeest!";
    dict["italia"] = "Buon Natale!";
    dict["libia"] = "Buon Natale!";
    dict["siria"] = "Milad Mubarak!";
    dict["marrocos"] = "Milad Mubarak!";
    dict["japao"] = "Merii Kurisumasu!";

    while (cin >> input) {
        it = dict.find(input);
        if (it == dict.end())
            cout << "--- NOT FOUND ---\n";
        else
            cout << it->second << "\n";
    }

    return 0;
}
